homeassistant/components/sensor/rova.py

```diff
@@ -0,0 +1,134 @@
+"""
+Support for Rova garbage calendar.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.rova/
+"""
+
+from datetime import datetime, timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_MONITORED_CONDITIONS, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['rova==0.0.1']
+
+# Config for rova requests.
+CONF_ZIP_CODE = 'zip_code'
+CONF_HOUSE_NUMBER = 'house_number'
+
+UPDATE_DELAY = timedelta(hours=12)
+SCAN_INTERVAL = timedelta(hours=12)
+
+# Supported sensor types:
+SENSOR_TYPES = {
+    'gft': ['GFT', 'mdi:recycle'],
+    'papier': ['Papier', 'mdi:recycle'],
+    'plasticplus': ['PMD', 'mdi:recycle'],
+    'rest': ['Rest', 'mdi:recycle']}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ZIP_CODE): cv.string,
+    vol.Required(CONF_HOUSE_NUMBER): cv.string,
+    vol.Optional(CONF_NAME, default='Rova'): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=['gft']):
+    vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Create the Rova data service and sensors."""
+    import rova.rova as r
+
+    zip_code = config[CONF_ZIP_CODE]
+    house_number = config[CONF_HOUSE_NUMBER]
+    name = config[CONF_NAME]
+
+    # Create new Rova object to  retrieve data
+    api = r.Rova(zip_code, house_number)
```

Comment:
Is there a way to validate the ZIP code and the house number?

Reply:
Good point. Added a check to see if ROVA collects at the given address.